ipaserver/install/dogtaginstance.py

```diff
@@ -275,6 +280,72 @@ def uninstall(self):
             logger.critical("failed to uninstall %s instance %s",
                             self.subsystem, e)
 
+    def __is_newer_tomcat_version(self, default=None):
+        try:
+            result = ipautil.run([paths.BIN_TOMCAT, "version"],
+                                 capture_output=True)
+            sn = re.search(
+                r'Server number:\s+([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)',
+                result.output)
+            if sn is None:
+                logger.info("tomcat version cannot be parsed, "
+                            "default to pre-%s", default)
+                return False
+            v = tasks.parse_ipa_version(sn.group(1))
+            if v >= tasks.parse_ipa_version(default):
+                return True
+        except ipautil.CalledProcessError as e:
+            logger.info(
+                "failed to discover tomcat version, "
+                "default to pre-%s, error: %s",
+                default, str(e))
+        return False
+
+    def secure_ajp_connector(self):
+        """ Update AJP connector to use a password protection  """
+
+        server_xml = lxml.etree.parse(paths.PKI_TOMCAT_SERVER_XML)
+        doc = server_xml.getroot()
+
+        # no AJP connector means no need to update anything
+        connectors = doc.xpath('//Connector[@port="8009"]')
+        if len(connectors) == 0:
+            return
+
+        # AJP connector is set on port 8009. Use non-greedy search to find it
+        connector = connectors[0]
+
+        # Detect tomcat version and choose the right option name
+        # pre-9.0.31.0 uses 'requiredSecret'
+        # 9.0.31.0 or later uses 'secret'
+        secretattr = 'requiredSecret'
+        oldattr = 'requiredSecret'
+        if self.__is_newer_tomcat_version('9.0.31.0'):
+            secretattr = 'secret'
+
+        rewrite = True
+        if secretattr in connector.attrib:
+            # secret is already in place
+            # Perhaps, we need to synchronize it with Apache configuration
+            self.ajp_secret = connector.attrib[secretattr]
+            rewrite = False
```

Comment:
If the server.xml contains a secret but ipa-pki-proxy.conf doesn't define secret, ipa-pki-proxy.conf won't get updated. Is it really the goal?